lib/galaxy/tools/parameters/output_collect.py

```diff
@@ -480,37 +480,48 @@ def discover_files(output_name, tool_provided_metadata, extra_file_collectors, j
             path = os.path.join(target_directory, filename)
             yield DiscoveredFile(path, extra_file_collector, JsonCollectedDatasetMatch(dataset, extra_file_collector, filename, path=path))
     else:
-        for (match, collector) in walk_over_extra_files(extra_file_collectors, job_working_directory, matchable):
+        for (match, collector) in walk_over_file_collectors(extra_file_collectors, job_working_directory, matchable):
             yield DiscoveredFile(match.path, collector, match)
 
 
-def discover_target_directory(extra_file_collector, job_working_directory):
-    directory = job_working_directory
-    if extra_file_collector.directory:
-        directory = os.path.join(directory, extra_file_collector.directory)
+def discover_target_directory(dir_name, job_working_directory):
+    if dir_name:
```

Comment:
This isn't equivalent to the previous version, and I guess this is why collecting outputs from tool provided metadata fails (see https://jenkins.galaxyproject.org/job/docker-framework/9815/).
In line 477 this function is called. I suppose changing the call in 477 to pass in `extra_file_collector.directory` instead of `extra_file_collector` may fix the failing tests.

Reply:
@mvdbeek Thanks for catching that - I'd missed your comment and I've just pushed in the fix as suggested. I changed it around a few times and must have forgotten to update the parameter.

Reply:
it seems to me this method may still have problems xref: https://github.com/galaxyproject/galaxy/issues/5560